src/errors/error_kind.rs

```diff
@@ -0,0 +1,319 @@
+use miette::Diagnostic;
+use thiserror::Error as ThisError;
+
+use crate::auth::AuthenticationScheme;
+use crate::StatusCode;
+
+/// Error kind for http-types
+#[derive(Debug, Diagnostic, ThisError)]
```

Comment:
Very minor nit, but I think this code would be marginally easier to understand if it just used `thiserror::Error` here.